performance-storage-service/pss_project/api/github_integration/check_run.py

```diff
@@ -0,0 +1,159 @@
+    if not check_run:
+        initialize_check_run(repo_client, commit_sha)
+        logger.debug('Check run did not exist. It has been initialized')
+
+
+def initialize_check_run(repo_client, sha):
+    """ Create the initial performance cop check run. The run starts in the
+    queued state """
+    create_body = create_initial_check_run(sha)
+    repo_client.create_check_run(create_body)
+    logger.debug('Initialized check run')
+
+
+def create_initial_check_run(sha):
+    """ Generate the body of the request to create the github check run. """
+    return {
+        "name": PERFORMANCE_COP_CHECK_NAME,
+        "head_sha": f"{sha}",
+        "status": "queued",
+        "output": {
+            "title": "Pending CI",
+            "summary": "This check will run after CI completes successfully"
+        }
+    }
+
+
+def complete_check_run(repo_client, commit_sha):
+    """ Update the check run with a complete status based on the performance
+    results. If the check run does not exist do nothing """
+    check_run = repo_client.get_commit_check_run_for_app(commit_sha, GITHUB_APP_IDENTIFIER)
+    if check_run:
+        repo_client.update_check_run(check_run.get('id'), performance_check_result())
+        logger.debug('Check run updated with performance results')
+
+
+def performance_check_result(branch):
+    """ Create a check run request body to make a check run as complete. Generate the status and output contents based
+    on the comparison between this PRs performance results and the nightly build's performance results."""
+    performance_comparisons = get_performance_comparisons(MASTER_BRANCH_NAME, branch)
+    conclusion = get_performance_comparisons_conclusion(performance_comparisons)
+    return {
+        "name": PERFORMANCE_COP_CHECK_NAME,
+        "status": "completed",
+        "conclusion": conclusion,
+        "output": {
+            "title": CONCLUSION_TITLE_MAP.get(conclusion),
+            "summary": CONCLUSION_SUMMARY_MAP.get(conclusion),
+            "text": generate_performance_result_markdown(performance_comparisons),
+        },
+    }
+
+
+def get_performance_comparisons(base_branch, change_branch):
+    """ Compare the performance results from two branches. This returns an array of tuples. The first item in the tuple
+    is the OLTPBench config and the second item in the tuple is the difference in throughput. """
+    result_comparisons = []
+    branch_results = OLTPBenchResult.get_latest_branch_results(change_branch)
+    master_results = OLTPBenchResult.get_branch_results_by_oltpbench_configs(base_branch, branch_results)
+    for b_result in branch_results:
+        for m_result in master_results:
+            if m_result.is_config_match(b_result):
+                config = b_result.get_test_config()
+                percent_diff = m_result.compare_throughput(b_result)
+                result_comparisons.append((config, percent_diff))
```

Comment:
Just want to check here, adding new fields to OLTPBench result in the future should be OK, but changing the current field names to the result JSON will cause the problem, right?

Reply:
yup correct

Reply:
Actually, the way it is setup is kinda nice. If the `throughput` field does not exist the % change will be 0 which will lead to a passing status.